apps/site/lib/site_web/controllers/schedule/line.ex

```diff
@@ -38,18 +40,86 @@ defmodule SiteWeb.ScheduleController.Line do
     Util.log_duration(__MODULE__, :do_call, [conn, opts])
   end
 
+  @spec get_updated_format_for_url_params(Plug.Conn.t()) :: Enum.t()
+  def get_updated_format_for_url_params(conn) do
+    new_params =
+      if Map.has_key?(conn.query_params, "direction_id") do
+        conn.query_params
+        |> Map.put("schedule_finder[direction_id]", Map.get(conn.query_params, "direction_id"))
+        |> Map.delete("direction_id")
+      else
+        conn.query_params
+      end
+
+    new_params =
+      if Map.has_key?(new_params, "origin") do
+        new_params
+        |> Map.put("schedule_finder[origin]", Map.get(new_params, "origin"))
+        |> Map.delete("origin")
+      else
+        new_params
+      end
+
+    new_params =
+      if Map.has_key?(new_params, "variant") do
+        new_params
+        |> Map.put("schedule_direction[variant]", Map.get(new_params, "variant"))
+        |> Map.delete("variant")
+      else
+        new_params
+      end
+
+    new_params
+  end
+
   def do_call(
-        %Conn{assigns: %{route: %Route{} = route, direction_id: direction_id}} = conn,
+        %Conn{
+          assigns: %{
+            route: %Route{} = route,
+            direction_id: direction_id
+          }
+        } = conn,
         args
       ) do
+    # check if URL has parameters like direction_id, origin or variant
+    # If so, they will get changed to:
+    # - "schedule_finder[direction_id]" and "schedule_finder[origin]" for direction_id and origin
+    # - "schedule_direction[variant]" for variant
+    # e.g. /schedules/31/line?direction_id=1&origin=525 will be changed to
+    # /schedules/31/line?schedule_finder%5Bdirection_id%5D=1&schedule_finder%5Borigin%5D=525
+
+    new_params = get_updated_format_for_url_params(conn)
+
+    if !Map.equal?(new_params, conn.query_params) do
```

Comment:
What am I missing here? why is this comparison giving an error in Dialyzer?

Reply:
@alicia-mbta I'll take a look after dinner